src/systemctl/systemctl.c

```diff
@@ -7336,6 +7340,49 @@ static int unit_file_create_new(
         if (r < 0)
                 return r;
 
+        if (unit_paths) {
+                _cleanup_strv_free_ char **appends = NULL;
+                _cleanup_free_ char *append = NULL;
+                _cleanup_fclose_ FILE *f = NULL;
+                char **path;
+
+                f = fopen(tmp_path, "ae");
+                if (!f)
+                        return log_error_errno(errno, "Failed to open \"%s\": %m", tmp_path);
+
+                strv_extendf(&appends, "\n\n%s\n", edit_discard_marker);
+
+                /* Add a comment with the contents of the original unit files */
+                STRV_FOREACH(path, unit_paths) {
+                        _cleanup_free_ char *contents = NULL;
+                        size_t size;
+
+                        /* Skip the file that's being edited */
+                        if (streq(*path, new_path))
+                                continue;
+
+                        r = strv_extendf(&appends, "# %s\n", *path);
+                        if (r < 0)
+                                return log_oom();
+
+                        r = read_full_file(*path, &contents, &size);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read \"%s\": %m", *path);
+
+                        r = strv_extend(&appends, contents);
+                        if (r < 0)
+                                return log_oom();
+                }
+
+                append = strv_join(appends, "\n");
+                if (!append)
+                        return log_oom();
+
+                r = fputs(append, f);
+                if (r == EOF)
+                        return log_error_errno(r, "Failed to write to \"%s\": %m", new_path);
```

Comment:
why concat this in memory first? you might as well just fputs() everything right away, it's more efficient since stdio concats the stuff anyway internally

Reply:
The concating is probably a carry-over from when I was making the contents of the units comments, so it's no longer necessary.
I'm in the process of converting all the string formatting to sequences of `fputs` instead. Is there any particular way to handle errors from `fputs` in this case? Searched for similar cases in the rest of the codebase, but they don't tend to bother checking the output of `fputs`. I'm currently only checking the result of the last call in each sequence.

Reply:
Call `fflush_and_check` at the end.